backend/amd64/simd_selection.ml

```diff
@@ -463,3 +463,347 @@ let () =
   Location.register_error_of_exn (function
     | Error err -> Some (Location.error_of_printer_file report_error err)
     | _ -> None)
+
+type width =
+  | W8
+  | W16
+  | W32
+  | W64
+
+type register =
+  | New of int
+  | Argument of int
+  | Result of int
+  | Original of int
+
+type vectorized_instruction =
+  { operation : Cfg.operation;
+    arguments : register array;
+    results : register array
+  }
+
+let vector_width = 128
+
+let vectorize_operation width (cfg_ops : Cfg.operation list) :
```

Comment:
(I think this function is too long
to be easy to understand, but
maybe it is enough to simply
move some nested functions.)

Reply:
what nested functions do you suggest to move out?